### DISCUSSION TITLE
Support of PUID and GUID

### DISCUSSION THREAD

**Discussion by Arnopello** [bpatrik/pigallery2#479] [Q&A]
Hi

I installed PiGallery2 on a Synology Nas, it's working like a charm. I was wondering as there seems to be no support for the PUID and GUID if it was using the root privileges? (I tested with a disabled account and configuration through environmental variables, PG2 was still loading photos) 
Then, wouldn't it be more secure to use an account with limited privileges to the data and actions permitted on the server?

Thanks for answers

**Comment by bpatrik** [comment]
The app itself does not control read/write rights, you need to enforce if from filesystem.

If you use docker to run, you would need to check how mounting the folder to docker changes rights. (I can images docker runs as root)

**Comment by Arnopello** [comment]
Thanks
So, if I get it right, if PiGallery2 happened to be an entrance door to my NAS for a hacker, it would only grant access to the mounted folder with the permissions I gave them (RO on my photos folder in my case, and the setup folders of PiGallery), and nothing more.

**Comment by bpatrik** [comment]
Hi yes.
I recommend using reverse proxy (like nginx) before pigallery and setting up the access rights in your OS. PiGallery does not provide any robust security in that sense. I think it is better to reduce the app rights, than me mocking some kind of security in javascript.

**Comment by Arnopello** [comment]
Thanks for your answers!
